File: Programming Skills/Valid Anagram/validAnagram.c
bool isAnagram(char* s, char* t) {
    if (strlen(s) != strlen(t)) {
        return false;
    }

    int counts[26] = {0};

    for (int i = 0; s[i] != '\0'; i++) {
        counts[s[i] - 'a']++;
        counts[t[i] - 'a']--;
    }

    for (int k = 0; k < 26; k++) {
        if (counts[k] != 0) {
            return false;
        }
    }

    return true;
}

File: Programming Skills/Valid Anagram/validAnagram.swift
class Solution {
    func isAnagram(_ s: String, _ t: String) -> Bool {
        guard s.count == t.count else { return false }

        var counts: [Character: Int] = [:]

        for char in s {
            counts[char, default: 0] += 1
        }

        for char in t {
            counts[char, default: 0] -= 1
        }

        return counts.values.allSatisfy { $0 == 0 }
    }
}